src/mscorlib/shared/System/String.Searching.cs

```diff
@@ -79,28 +82,77 @@ public unsafe int IndexOf(char value, int startIndex, int count)
 
```

Comment:
nit: reduce the comparisons in the bounds checks:

`if (startIndex < 0 || startIndex > Length)` => `if ((uint)startIndex > (uint)Length)`

`if (count < 0 || count > Length - startIndex)` => `if ((uint)count > (uint)(Length - startIndex))`